### DISCUSSION TITLE
Conference talk topics

### DISCUSSION THREAD

**Discussion by OndroMih** [piranhacloud/piranha#2589] [Show and tell]
Hi,

I'm thinking about submitting some talk about Piranha to a few conferences. The CFP is currently open for [EclipseCon](https://www.eclipsecon.org/2022) in Germany (until 15 June) and [Devoxx](https://devoxx.be/) Belgium (until 15th July).

For EclipseCon, I'm thinking about submitting something about Jakarta EE 10, GlassFish and Piranha as a cloud alternative to GlassFish built on many Eclipse components used by GlassFish. I might also submit a talk just about Piranha as it implements Jakarta EE and thus is still related to Eclipse Foundation. But I'm not sure I'll have enough time to think about an abstract just about Piranha as I'm very new to the project.

For Devoxx, I'd definitely like to submit a talk just about Piranha to present it. But I'd like to put it in some context and focus on something that's appealing to people. For example fast boot time, or cold starts in AWS Lambda, or developing microservices and serverless apps affectively with Java.

Do you have any ideas? Or even better, would anyone like to co-submit a talk with me?

**Comment by mnriem** [comment]
Go for it! The most minimal is just using the 3 core JARs (core directory) and an HTTP engine (pick one in the http subdirectory) which then would make it even smaller than Piranha Nano.

**Comment by OndroMih** [comment]
I have this topic so far. With this, I'd like to explain Serverless, cold starts, costs and show how Java is fast, what slows it down and how modern frameworks like Quarkus and Piranha, can help, including GraalVM native compilation, although that's not necessary in most cases.

Any comments?

Title: Afraid of Java cold starts in Serverless? Fear not, Jakarta EE is super fast!

For years, we’ve been told that Java suffers from cold starts in AWS Lambda and Serverless in general. Believe it not. Java is extremely fast to start, the simplest Java program starts in milliseconds. It’s the Java frameworks and libraries that slow things down in general.

In this session, I want to show you the truth about Serverless Java and cold starts, what’s slow, what’s fast and why it matters. We’ll go through what a cold start is, how cold starts in AWS Lambda impact your application responsiveness and cloud costs, and what are the options to optimize both. We’ll explore and compare Quarkus and Piranha Cloud, the most modern and fastest Jakarta EE frameworks. We’ll test the limits of what they are capable of, how fast they start and whether we even need to think about any optimizations. Then I’ll explain what optimizations may help if you need to go even further, like tuning the JVM, using a custom JVM runtime or even building a GraalVM native image from a Java application. With those, we can decrease the Java cold start to the absolute minimum but with a few limitations and cost implications.

I’ll show and tell, with very few slides. You’ll see live for yourselves how Java AWS Lambda functions built with Jakarta EE can start fast and decrease the cold starts to the absolute minimum which you can barely notice.
